timemachine/fe/rbfe.py

```diff
@@ -798,7 +796,11 @@ def run_edges_parallel(
             file_client,
         )
         for edge_idx, edge in enumerate(edges)
-    )
+    ]
+
+    paths = []
+    while jobs:
+        job = jobs.pop(0)
+        paths.append(job.result())
 
-    paths = [job.result() for job in jobs]
```

Comment:
Curiosity / nit: is this second change necessary?

(Would have thought the above change from lazy `( ... )` -> eager `[ ... ]` is sufficient to ensure all edge jobs are enqueued before we start waiting on results, don't immediately see the necessity for this additional change.)

Reply:
Oh, right 🤦 , not required for the parallelism fix but required to remove references to jobs after completion / keep memory usage from growing with `len(job)` -- sorry, had fixated on just one part of the PR description. This looks like a clean way to accomplish both goals. 👍

Reply:
Added a comment in 14e2764 to clarify the motivation for this.